Parse/src/test/java/com/parse/ParseInstallationTest.java

```diff
@@ -109,6 +111,50 @@ public void testImmutableKeys() {
   }
 
   @Test
+  public void testSaveAsync() throws Exception {
+    String sessionToken = "sessionToken";
+    Task<Void> toAwait = Task.forResult(null);
+
+    ParseCurrentInstallationController controller =
+        mock(ParseCurrentInstallationController.class);
+    ParseInstallation currentInstallation = new ParseInstallation();
+    when(controller.getAsync())
+        .thenReturn(Task.forResult(currentInstallation));
+    ParseCorePlugins.getInstance()
+        .registerCurrentInstallationController(controller);
+    ParseObject.State state = new ParseObject.State.Builder("_Installation")
+        .put("deviceToken", "deviceToken")
+        .build();
+    ParseInstallation installation = ParseInstallation.getCurrentInstallation();
+    assertNotNull(installation);
+    installation.setState(state);
+
+    ParseObjectController objController = mock(ParseObjectController.class);
+    // mock return task when Installation was deleted on the server
+    Task<ParseObject.State> taskError = Task.forError(new ParseException(ParseException.OBJECT_NOT_FOUND, ""));
+    // mock return task when Installation was re-saved to the server
+    Task<ParseObject.State> task = Task.forResult(null);
+    when(objController.saveAsync(
+        any(ParseObject.State.class),
+        any(ParseOperationSet.class),
+        eq(sessionToken),
+        any(ParseDecoder.class)))
+        .thenReturn(taskError)
+        .thenReturn(task);
+    ParseCorePlugins.getInstance()
+        .registerObjectController(objController);
+
+    installation.saveAsync(sessionToken, toAwait);
+
+    verify(controller, times(1)).getAsync();
```

Comment:
minor optimization -- this can be `verify(controller).getAsync();```

```Alias to verify(mock, times(1)) E.g:


   verify(mock).someMethod("some arg");
 
Above is equivalent to:

   verify(mock, times(1)).someMethod("some arg");
 
```